### ISSUE TITLE
Where shall we keep re-usables?

### ISSUE THREAD

**Issue by EdwardAndress** [moj-analytical-services/our-coding-standards#20]
If we write some nice functions that perform useful tasks which we might want to use again and share with other people (for example, I wrote something at the weekend which extracts the names of the most significant features from a regression model and trains a new model using only those features*), where should we keep them?  Should they all be in one repo?  Or one for each language we're using?  How should the functions be name spaced within that repo?  Are they each in their own package?

* I will not be the slightest bit surprised if someone points out that this has already been done by someone else! ;)

**Comment by RobinL** [comment]
I guess the hard problem for the user is being able to find and understand the function if it's one among a long list.  You can search within the code of a repo.  Could it be a single repo with one file for each function (or group of functions), which are properly documented using either roxygen/docstring (depending on language), and which have unit tests at the bottom of that file?

**Comment by EdwardAndress** [comment]
Yep, documentation is likely to be key when it comes to finding things.  Peer review will also help to inform the team about what's been created.  Ideally, you'd always be within earshot of someone who already knows about the function you need.  We can also get people to present their re-usables at the DaSHion show or lunch and code.
